src/scaffoldmaker/meshtypes/meshtype_3d_brainstem.py

```diff
@@ -137,103 +288,311 @@ def generateBaseMesh(region, options):
+                [[0.0, 0.0,  0.0], [0.0, 0.0, 1.8], [0.0, 3.6, 0.0], [0.0, 0.0, 0.0], [-3.6, 0.0, 0.0], [0.0, 0.0, 0.0]],
+                [[0.0, 0.0,  1.8], [0.0, 0.0, 1.8], [0.0, 3.6, 0.0], [0.0, 0.0, 0.0], [-3.6, 0.0, 0.0], [0.0, 0.0, 0.0]],
+                [[0.0, 0.0,  3.6], [0.0, 0.0, 1.8], [0.0, 3.6, 0.0], [0.0, 0.0, 0.0], [-3.6, 0.0, 0.0], [0.0, 0.0, 0.0]],
+                [[0.0, 0.0,  5.3], [0.0, 0.0, 1.8], [0.0, 3.6, 0.0], [0.0, 0.0, 0.0], [-3.6, 0.0, 0.0], [0.0, 0.0, 0.0]],
+                [[0.0, 0.0,  7.1], [0.0, 0.0, 1.8], [0.0, 3.6, 0.0], [0.0, 0.0, 0.0], [-3.6, 0.0, 0.0], [0.0, 0.0, 0.0]],
+                [[0.0, 0.0,  8.9], [0.0, 0.0, 1.8], [0.0, 3.6, 0.0], [0.0, 0.0, 0.0], [-3.6, 0.0, 0.0], [0.0, 0.0, 0.0]],
+                [[0.0, 0.0, 10.7], [0.0, 0.0, 1.8], [0.0, 3.6, 0.0], [0.0, 0.0, 0.0], [-3.6, 0.0, 0.0], [0.0, 0.0, 0.0]]
+            ]
+
+            cylinderCentralPath1 = CylinderCentralPath(tmp_fm, centralPath1, elementsCountAlong, cylinderScaffold=False, annotationGroup=brainTermsAlong)
+            base1 = CylinderEnds(elementsCountAcrossMajor, elementsCountAcrossMinor,
+                                centre=[0.0, 0.0, 0.0],
+                                alongAxis=cylinderCentralPath1.alongAxis[0],
+                                majorAxis=cylinderCentralPath1.majorAxis[0],
+                                minorRadius=cylinderCentralPath1.minorRadii[0])
+
+            cylinder2 = CylinderMesh(tmp_fm, organ_coordinates, elementsCountAlong, base1,
+                                     cylinderShape=cylinderShape,
+                                     cylinderCentralPath=cylinderCentralPath1,
+                                     useCrossDerivatives=False)
+
+            sir = tmp_region.createStreaminformationRegion()
+            srm = sir.createStreamresourceMemory()
+            tmp_region.write(sir)
+            result, buffer = srm.getBuffer()
+            sir = region.createStreaminformationRegion()
+            srm = sir.createStreamresourceMemoryBuffer(buffer)
+            fielditer = fm.createFielditerator()
+            field = fielditer.next()
+            while field.isValid():
+                field = fielditer.next()
+
+            # Annotating groups
+            iRegionBoundaries = [int(7 * elementsCountAlong / 15), int(13 * elementsCountAlong / 15)]
+            for elementIdentifier in range(1, mesh.getSize() + 1):
+                element = mesh.findElementByIdentifier(elementIdentifier)
+                brainstemMeshGroup.addElement(element)
+                if elementIdentifier > (iRegionBoundaries[-1] * elementsPerLayer):
+                    midbrainMeshGroup.addElement(element)
+                elif (elementIdentifier > (iRegionBoundaries[0] * elementsPerLayer)) and (elementIdentifier <= (iRegionBoundaries[-1] * elementsPerLayer)):
+                    ponsMeshGroup.addElement(element)
+                else:
+                    medullaMeshGroup.addElement(element)
```

Comment:
The field named "brainstem_coordinates" (not organ_coordinates) should be defined the same for all species: 10 units long 2-4-4 division into midbrain-pons-medulla with radius 1 unit. I can't see these coordinates in the output file -- needs to be written out.

Reply:
I changed it to 2-3-3.